src/MSBuild.UnitTests/XMake_Tests.cs

```diff
@@ -1259,7 +1259,38 @@ private void RunPriorityBuildTest(ProcessPriorityClass expectedPrority, params s
             logContents.ShouldContain(expected, () => logContents);
         }
 
-#region IgnoreProjectExtensionTests
+        /// <summary>
+        /// Test the case where there is one solution file, one solution filter file, and one project file.
+        /// </summary>
+        [Fact]
+        public void TestOneProjectOneSolutionFileOneSolutionFilterFile()
+        {
+            string[] projects = new string[] { "my.proj", "my.sln", "my.slnf" };
+            string[] extensionsToIgnore = Array.Empty<string>();
+            IgnoreProjectExtensionsHelper projectHelper = new IgnoreProjectExtensionsHelper(projects);
+            MSBuildApp.ProcessProjectSwitch(Array.Empty<string>(), extensionsToIgnore, projectHelper.GetFiles).ShouldBe("my.sln", StringCompareShould.IgnoreCase);
+        }
+
+        /// <summary>
+        /// Test the default file to build in cases involving at least one solution filter file.
+        /// </summary>
+        [Theory]
+        [InlineData("my.proj my.sln my.slnf my.sln")]
+        [InlineData("abc.proj bcd.csproj slnf.slnf other.slnf abc.proj")]
+        [InlineData("abc.sln slnf.slnf abc.slnf abc.sln")]
+        [InlineData("abc.csproj abc.slnf not.slnf abc.csproj")]
+        [InlineData("abc.slnf abc.slnf")]
+        public void TestDefaultBuildWithSolutionFilter(string projectData)
```

Comment:
nit: you can use a string array for the input and another string for the expected output and have them as two separate parameters to the test method to avoid splitting and inferring what the last split means.

Reply:
I actually poked at that a little but had trouble. Is there a good example of that syntax? It didn't like:
`[InlineData(new string[] { ... })]`

Reply:
Found an example: [https://github.com/dotnet/msbuild/blob/master/src/Build.UnitTests/Evaluation/SemiColonTokenizer_Tests.cs#L24](https://github.com/dotnet/msbuild/blob/master/src/Build.UnitTests/Evaluation/SemiColonTokenizer_Tests.cs#L24)

---
In reply to: [494614444](https://github.com/dotnet/msbuild/pull/5740#discussion_r494614444) [](ancestors = 494614444)